### PR TITLE
Single Track model with Steering-Rate Inputs

### PR DESCRIPTION
**Pull request by patrickhart**
Added a SingleTrackModel that takes the acceleration and steering-rate as input.

### CODE REVIEW

File: bark/models/dynamic/single_track_steering_rate.hpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  State StateSpaceModel(const State& x, const Input& u) const {
+    State tmp(static_cast<int>(StateDefinition::MIN_STATE_SIZE ) + 1);
+    tmp << 1,
+        x(StateDefinition::VEL_POSITION) *
+            cos(x(StateDefinition::THETA_POSITION)),
+        x(StateDefinition::VEL_POSITION) *
+            sin(x(StateDefinition::THETA_POSITION)),
+        x(StateDefinition::VEL_POSITION) * tan(
+          x[5]) / wheel_base_,
```
Comment (klemense1) [inline bark/models/dynamic/single_track_steering_rate.hpp:40]: x[5] should also use StateDefinition

File: bark/models/dynamic/single_track_steering_rate.hpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#include "bark/commons/transformation/frenet_state.hpp"
+#include "bark/models/dynamic/dynamic_model.hpp"
+
+namespace bark {
+namespace models {
+namespace dynamic {
+
+
+class SingleTrackSteeringRateModel : public DynamicModel {
+ public:
```
Comment (klemense1) [inline bark/models/dynamic/single_track_steering_rate.hpp:23]: please add some doxygen documentation here... it is not obvious why the model consists of...

File: bark/models/tests/behavior_dynamic_model.cc [hunk 3]
```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2020 fortiss GmbH
```
Comment (klemense1) [inline bark/models/tests/behavior_dynamic_model.cc:1]: rename the file test_...?

**Review by klemense1 — CHANGES_REQUESTED** [review]
Looks good, please follow up on my remarks.

File: bark/models/dynamic/single_track_steering_rate.hpp (hunk 1 above)
Comment (patrickhart) [reply bark/models/dynamic/single_track_steering_rate.hpp:40]: The problem is that at the moment `StateDefinition::MIN_STATE_SIZE` is equal to `5`. So I don't think we can change that with major refactoring.

Comment (klemense1) [reply bark/models/dynamic/single_track_steering_rate.hpp:40]: Okay, although I don't like it :-)

**Review by klemense1 — APPROVED** [review]

**Pull request merged**